src/Psalm/Internal/PhpVisitor/Reflector/TypeHintResolver.php

```diff
@@ -55,6 +74,52 @@ public static function resolve(
             return $type;
         }
 
+        if ($hint instanceof PhpParser\Node\IntersectionType) {
+            $type = null;
+
+            if (!$hint->types) {
+                throw new UnexpectedValueException('bad');
```

Comment:
Please give a proper text to the Exception. I know those are not supposed to happen, but when they end up happening anyway, it's tiresome to find the correct one when it just says "bad" :)

Reply:
That's called lazy copy-pasting! Will do.